src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/autocaseupdate/AutoCaseUpdater.java

```diff
@@ -43,18 +45,36 @@ public AutoCaseUpdater(
 
     public AutoCaseUpdateResult updateCase(Envelope envelope) {
         try {
-            List<Long> matchingCases = ccdApi.getCaseRefsByEnvelopeId(envelope.id, envelope.container);
-
-            switch (matchingCases.size()) {
-                case 1:
-                    String caseId = String.valueOf(matchingCases.get(0));
-                    return handleSingleMatchingCase(caseId, envelope);
-                case 0:
-                    return handleNoMatchingCases(envelope);
-                default:
-                    return handleMultipleMatchingCases(matchingCases, envelope);
-            }
-        } catch (Exception exc) {
+            return caseFinder
+                .findCase(envelope)
+                .map(
+                    existingCase -> {
```

Comment:
Don't we have to handle the scenario where multiple cases found for the envelope?

Reply:
I just saw your comment above.